src/main/kotlin/com/demonwav/mcdev/platform/fabric/framework/FabricPresentationProvider.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2020 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.platform.fabric.framework
+
+import com.demonwav.mcdev.asset.PlatformAssets
+import com.demonwav.mcdev.util.localFile
+import com.intellij.framework.library.LibraryVersionProperties
+import com.intellij.openapi.roots.libraries.LibraryPresentationProvider
+import com.intellij.openapi.vfs.VirtualFile
+import java.util.jar.JarFile
+
+class FabricPresentationProvider : LibraryPresentationProvider<LibraryVersionProperties>(FABRIC_LIBRARY_KIND) {
+
+    override fun getIcon(properties: LibraryVersionProperties?) = PlatformAssets.FABRIC_ICON
+
+    override fun detect(classesRoots: MutableList<VirtualFile>): LibraryVersionProperties? {
+        for (classesRoot in classesRoots) {
+            if (classesRoot.name.endsWith(".jar")) {
+                JarFile(classesRoot.localFile).runCatching {
```

Comment:
You may want to catch IOExceptions, I did this mistake for Velocity and look at all those fcking issues reported. (JarFile instanciation may throw too)

Reply:
Oops, that's what the `runCatching` was for, but I forgot that the constructor also throws the exception.